### PR TITLE
feat: add guide on federated search

### PR DESCRIPTION
**Pull request by rfazeli**
This PR adds a guide on performing federated search across multiple indices on an Elasticsearch cluster

Signed off by: reza@ibm.com

### CODE REVIEW

File: integrations/extensions/docs/elasticsearch-install-and-setup/federated_search.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Federated Search in Elasticsearch
+
+This guide dives into performing a single search query across multiple indexes within an Elasticsearch cluster. It assumes these indexes share the same relevant fields for your search results.
+
+## Methods for Multi-Index Search
+
+There are several approaches to searching across multiple indexes in Elasticsearch:
+
+### 1. Comma-Separated Indices:
```
Comment (zach-shu) [inline integrations/extensions/docs/elasticsearch-install-and-setup/federated_search.md:9]: Thanks, Reza! I think another option is to use an alias where multiple indices share the same alias.

Comment (rfazeli) [reply integrations/extensions/docs/elasticsearch-install-and-setup/federated_search.md:9]: Thanks, Zach! That's right. I can add that too. The only reason I didn't was to keep it simple as that requires an extra step of creating the alias as well.

File: integrations/extensions/docs/elasticsearch-install-and-setup/federated_search.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+For similar functionality with WatsonX Assistant's built-in search extension, follow the [official Elasticsearch search integration set up documentation](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-search-elasticsearch-add). The key difference is specifying your `Elasticsearch index` field using one of the methods described above. For example:
+
+<img src="./assets/federated_search_official_search_extension.png" width="838" height="547" />
+
+### Custom Extension
+
+When building a custom extension in WatsonX Assistant with the Elasticsearch API, you can leverage these same index name patterns when setting the `index_name` parameter. Follow the guide on [building a custom extension in watsonx Assistant with Elasticsearch API](../../starter-kits/elasticsearch/README.md#build-a-custom-extension-in-watsonx-assistant-with-elasticsearch-api).
+
+Then follow one of the methods listed above when setting the `index_name` parameter to search across multiple indices within your Elasticsearch cluster. For example:
+<img src="./assets/federated_search_elasticsearch_custom_extension.png" width="669" height="627" />
```
Comment (zach-shu) [inline integrations/extensions/docs/elasticsearch-install-and-setup/federated_search.md:139]: The image needs to be in a new line.

**Review by zach-shu — APPROVED** [review]
LGTM. Thanks, Reza!

**Pull request merged**